src/main/java/io/aiven/kafka/connect/opensearch/OpensearchSinkConnectorConfig.java

```diff
@@ -574,6 +579,39 @@ public DocumentIDStrategy docIdStrategy(final String topic) {
                 : DocumentIDStrategy.RECORD_KEY;
     }
 
+    public Function<String, String> topicToIndexNameConverter() {
+        return topic -> dataStreamEnabled()
+                ? convertTopicToDataStreamName(topic)
+                : convertTopicToIndexName(topic);
+    }
+
+    private String convertTopicToIndexName(final String topic) {
```

Comment:
Could be `static`, right?

Reply:
good question i do not see the reason tbh

Reply:
No need to capture the instance in lamda expression?